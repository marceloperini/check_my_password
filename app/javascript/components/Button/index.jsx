import PropTypes from 'prop-types';
import React from 'react';

const Button = (props) => {
  const { children, onClick, className, type, color } = props;

  return (
    <button
      type={type}
      className={`btn btn-${color} ${className}`}
      onClick={onClick}>
      {children}
    </button>
  );
};

Button.defaultProps = {
  onClick: () => {},
  className: '',
};

Button.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
  className: PropTypes.string,
  type: PropTypes.oneOf([
    'button',
    'submit',
  ]).isRequired,
  color: PropTypes.oneOf([
    'primary',
    'secondary',
    'success',
    'danger',
    'warning',
    'info',
    'light',
    'dark',
    'link',
  ]).isRequired,
};

export default Button;
