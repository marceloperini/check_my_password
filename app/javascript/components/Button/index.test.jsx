import React from 'react';
import renderer from 'react-test-renderer';

import Button from '.';

describe('Button', () => {
  const type = 'button';
  const color = 'primary';

  test('when children are seted', () => {
    const component = renderer.create(
      <Button
        type={type}
        color={color}
      >
        <div></div>
      </Button>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('when onClick are seted', () => {
    const onClick = () => {};
    const component = renderer.create(
      <Button
        type={type}
        color={color}
        onClick={onClick}
      >
        <div></div>
      </Button>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('when className are seted', () => {
    const component = renderer.create(
      <Button
        type={type}
        color={color}
        className="NewClassName"
      >
        <div></div>
      </Button>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('when optional props are not seted', () => {
    const component = renderer.create(
      <Button
        type={type}
        color={color}
      >
        <div></div>
      </Button>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
