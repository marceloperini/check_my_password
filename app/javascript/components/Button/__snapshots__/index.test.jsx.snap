// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when children are seted 1`] = `
<button
  className="btn btn-primary "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div />
</button>
`;

exports[`Button when className are seted 1`] = `
<button
  className="btn btn-primary NewClassName"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div />
</button>
`;

exports[`Button when onClick are seted 1`] = `
<button
  className="btn btn-primary "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div />
</button>
`;

exports[`Button when optional props are not seted 1`] = `
<button
  className="btn btn-primary "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div />
</button>
`;
